### PR TITLE
[CORE-652] add bridge disabling to prepareProposal, processProposal, acknowledgeBridges, and completeBridge

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
If bridging is disabled
* do not propose bridge events (prepare)
* reject non-empty bridge events in process (process)
* do not acknowledge bridge events (deliverTx)
* do not complete bridge events

### Test Plan
unit testing and integration testing in this PR
local testing (through running localnet)

### Author/Reviewer Checklist
- [ ] If this change affects functionality (features, bug fixes, breaking changes, etc.), update `protocol/CHANGELOG.md` and/or `indexer/CHANGELOG.md` appropriately.
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### Summary by CodeRabbit

- New Feature: Added a validation check to prevent bridging operations when bridging is disabled. This includes operations such as acknowledging bridges and completing bridges.
- New Feature: Introduced a new error `ErrBridgingDisabled` to indicate when bridging operations are attempted while bridging is disabled.
- Test: Enhanced test coverage to include scenarios where bridging is disabled. This includes tests for acknowledging bridges, processing proposals, and completing bridges.
- Refactor: Updated the `ProcessBridgeKeeper` interface and its mock to include a new method `GetSafetyParams` for retrieving safety parameters, including the bridging disabled status.

### CODE REVIEW

File: protocol/app/process/transactions_test.go [hunk 1]
```diff
@@ -401,6 +431,12 @@ func TestProcessProposalTxs_Validate_Valid(t *testing.T) {
 			indexPriceCache.UpdatePrices(constants.AtTimeTSingleExchangePriceUpdate)
 
 			mockBridgeKeeper := &mocks.ProcessBridgeKeeper{}
+			mockBridgeKeeper.On("GetSafetyParams", mock.Anything).Return(
+				bridgetypes.SafetyParams{
+					IsDisabled:  tc.bridgingDisabled,
+					DelayBlocks: 5, // dummy value, not considered by Validate.
```
Comment (clemire) [inline protocol/app/process/transactions_test.go:437]: Thanks for these comments!

File: protocol/x/bridge/keeper/acknowledge_bridges.go [hunk 2]
```diff
@@ -20,6 +20,14 @@ func (k Keeper) GetAcknowledgeBridges(
 	ctx sdk.Context,
 	blockTimestamp time.Time,
 ) (msg *types.MsgAcknowledgeBridges) {
+	// Do not propose bridge events if bridging is disabled.
+	safetyParams := k.GetSafetyParams(ctx)
```
Comment (clemire) [inline protocol/x/bridge/keeper/acknowledge_bridges.go:24]: meganit, totally optional:
`if k.GetSafetyParams(ctx).IsDisabled {`

(I think you follow this elsewhere where you can.)

File: protocol/x/bridge/keeper/acknowledge_bridges_test.go [hunk 3]
```diff
@@ -59,16 +71,31 @@ func TestAcknowledgeBridges(t *testing.T) {
...
+
+				// Verify that no messages were delayed.
+				mockDelayMsgKeeper.AssertNotCalled(t, "DelayMessageByBlocks")
 
-			// Assert expected AcknowledgedEventInfo.
-			aei := bridgeKeeper.GetAcknowledgedEventInfo(ctx)
-			require.Equal(t, tc.expectedAEI, aei)
+				// Verify that AcknowledgedEventInfo was not updated.
+				require.Equal(t, initialAei, bridgeKeeper.GetAcknowledgedEventInfo(ctx))
+			} else { // briding is not disabled.
```
Comment (clemire) [inline protocol/x/bridge/keeper/acknowledge_bridges_test.go:88]:
```suggestion
			} else { // bridging is not disabled.
```

File: protocol/x/bridge/keeper/acknowledge_bridges_test.go [hunk 4]
```diff
@@ -59,16 +71,31 @@ func TestAcknowledgeBridges(t *testing.T) {
...
-			require.NoError(t, err)
+			// Invoke AcknowledgeBridges.
+			err = bridgeKeeper.AcknowledgeBridges(ctx, tc.bridgeEvents)
+
+			if tc.bridgingDisabled { // bridging is disabled.
+				// Verify that AcknowledgeBridges returns ErrBridgingDisabled.
+				require.ErrorIs(t, err, types.ErrBridgingDisabled)
+
+				// Verify that no messages were delayed.
+				mockDelayMsgKeeper.AssertNotCalled(t, "DelayMessageByBlocks")
```
Comment (clemire) [inline protocol/x/bridge/keeper/acknowledge_bridges_test.go:84]: +1, nice

File: protocol/x/bridge/keeper/acknowledge_bridges_test.go [hunk 5]
```diff
@@ -204,6 +232,29 @@ func TestGetAcknowledgeBridges(t *testing.T) {
 				Events: []types.BridgeEvent{},
 			},
 		},
+		"No event is proposed when bridging is diabled": {
```
Comment (clemire) [inline protocol/x/bridge/keeper/acknowledge_bridges_test.go:235]:
```suggestion
		"No event is proposed when bridging is disabled": {
```

**Review by clemire — APPROVED** [review]

**Review by roy-dydx — APPROVED** [review]

**Pull request merged**